packages/apps/plugins/plugin-client/src/ClientPlugin.tsx

```diff
@@ -31,23 +30,41 @@ export const parseClientPlugin = (plugin?: Plugin) =>
   (plugin?.provides as any).client instanceof Client ? (plugin as Plugin<ClientPluginProvides>) : undefined;
 
 export const ClientPlugin = ({
-  debugIdentity,
   types,
   appKey,
   ...options
 }: ClientPluginOptions): PluginDefinition<{}, ClientPluginProvides> => {
+  let client: Client;
+
   // TODO(burdon): Document.
   registerSignalFactory();
 
-  const client = new Client({ config: new Config(Envs(), Local(), Defaults()), ...options });
-
   return {
     meta,
     initialize: async () => {
       let error: unknown = null;
 
+      client = new Client({ config: new Config(await Envs(), Local(), Defaults()), ...options });
+
+      const oldClient = new Client({
```

Comment:
disable shell in old client?

Reply:
Feels like the migration code should be in its own function/file

Reply:
Ideally the migration code would be separate, but if the old client isn't initialized first then the global hooks are registered to the wrong client. This code will get removed after a couple months so not the end of the world.